receiver/splunkhecreceiver/splunk_to_logdata.go

```diff
@@ -53,7 +54,12 @@ func SplunkHecToLogData(logger *zap.Logger, events []*splunk.Event, resourceCust
 
 		// Splunk timestamps are in seconds so convert to nanos by multiplying
 		// by 1 billion.
-		logRecord.SetTimestamp(pdata.TimestampUnixNano(event.Time * 1e9))
+		if event.Time == nil {
+			logger.Debug("Unsupported event time", zap.Any("value", event.Event))
+			return ld, errors.New(nilTimestamp)
+		}
```

Comment:
The `splunkhecreceiver` needs to be changed as it uses the same model. I wasn't sure of what validation you guys would want so I've done something basic here. Previously the timestamps coming into the receiver presumably all had a value so perhaps this is not the best thing to do?

Reply:
From examples here (https://docs.splunk.com/Documentation/Splunk/8.0.6/Data/HECExamples) looks like time is an optional field, always.
We should accept events with missing time field and just pass them as best we can, without setting the timestamp in that case.

Reply:
Ok thanks - I'll fix that when I get sometime this week

Reply:
So based on what @atoulme says, I assume this `if` is not needed? If it is a valid situation for the time to be missing why are we erroring on it?

Reply:
Thanks - that was an oversight by me. I've now adjusted this code. Now the behaviour is that in the event of `event.Time == nil`, we won't set a timestamp on the log records. That means they'll get their default value (0). I don't think there's a way to set the log record timestamp to be nil/empty

Reply:
looks good to me, and if there was any confusion here, that's on me for doing a bad job the first time around.